includes/Checker/With_Shared_Preparations.php

```diff
@@ -0,0 +1,25 @@
+<?php
+/**
+ * Interface WordPress\Plugin_Check\Checker\With_Shared_Preparations
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+/**
+ * Interface for the shared preparations.
+ *
+ * @since n.e.x.t
+ */
+interface With_Shared_Preparations {
+
+	/**
+	 * Get list of preparation classes.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array Returns an array of preparation class names.
```

Comment:
I think we need the return description here to expand a bit more: This is not captured in the ACs, so apologies for missing that. @jjgrainger and I had talked about that we also need to support a way to provide constructor arguments here. For example, if one of the class names requires constructor arguments, the consuming code has to know which ones should be used.

So I think we should maybe make the expected return value here a map of class name => constructor arguments pairs. If the class does not need any constructor arguments, it would just be an empty array.
```suggestion
	 * @return array Returns a map of $class_name => $constructor_args pairs.
```

Reply:
@felixarntz I updated this now and also added these details in the function doc comment. Thank you!